### PR TITLE
Refactor execute method in http

### PR DESCRIPTION
**Pull request by ShrutiVerma97**

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 1]
```diff
@@ -369,7 +369,31 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
...
+          case (self, _)                                 => self
+        }
+
+      case FoldHttp(self, ee, bb, dd) =>
+        self.execute2(a).foldExit(ee(_).execute2(a), bb(_).execute2(a), dd.execute2(a))
+
+      case RunMiddleware(app, mid) => mid(app).execute2(a)
+    }
+  }
+  final private[zhttp] def execute(a: A): HExit[R, E, B]                                       =
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Http.scala]: You can modify the current execute also. I don't think you need a fresh copy of the implementation

File: zio-http/src/main/scala/zhttp/http/HttpConverter.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        override def getHeaders: Headers = Headers.make(jReq.headers())
+
+        override private[zhttp] def getBodyAsByteBuf: Task[ByteBuf] = Task(jReq.content())
+
+        override def remoteAddress: Option[InetAddress] = ???
+      }
+    }
+  }
+
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/HttpConverter.scala]: Add an identity convertor also 

```scala

// not implicit
private val empty: HttpConvertor[Any, Nothing] = new HttpConvertor[Any, Nothing] {
  override def convert(a: Any): Nothing = a
}

implicit def identity[A]: HttpConvertor[A, A] = empty // Create empty once for performance.

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 3]
```diff
@@ -369,28 +369,31 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
    * should be stack safe. The performance improves quite significantly if no additional heap allocations are required
    * this way.
    */
-  final private[zhttp] def execute(a: A): HExit[R, E, B] =
+
+  final private[zhttp] def execute[X](a: X, b: => (X => A)): HExit[R, E, B] = {
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Http.scala]: Keep this as implicit.

**Comment by ShrutiVerma97** [comment]
Optimisations done for a very small use case. Closing it for now.

**Pull request closed without merging**